tools/src/main/java/org/apache/kafka/tools/consumer/TransactionLogMessageFormatter.java

```diff
@@ -0,0 +1,77 @@
+ */
+package org.apache.kafka.tools.consumer;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.MessageFormatter;
+import org.apache.kafka.common.protocol.ByteBufferAccessor;
+import org.apache.kafka.coordinator.transaction.generated.TransactionLogKey;
+import org.apache.kafka.coordinator.transaction.generated.TransactionLogValue;
+import org.apache.kafka.coordinator.transaction.generated.TransactionLogValueJsonConverter;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.ByteBuffer;
+import java.util.Optional;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class TransactionLogMessageFormatter implements MessageFormatter {
+
+    @Override
+    public void writeTo(ConsumerRecord<byte[], byte[]> consumerRecord, PrintStream output) {
+        Optional.ofNullable(consumerRecord.key())
+                .map(key -> readToTransactionLogKey(ByteBuffer.wrap(key)))
+                .ifPresent(transactionLogKey -> {
+                    short version = ByteBuffer.wrap(consumerRecord.key()).getShort();
+                    ObjectNode json = new ObjectNode(JsonNodeFactory.instance);
+                    json.set("version", new TextNode(Short.toString(version)));
+                    
+                    if (version >= TransactionLogValue.LOWEST_SUPPORTED_VERSION
+                            && version <= TransactionLogValue.HIGHEST_SUPPORTED_VERSION) {
+                        byte[] value = consumerRecord.value();
+                        TransactionLogValue transactionLogValue = 
+                                new TransactionLogValue(new ByteBufferAccessor(ByteBuffer.wrap(value)), version);
+                        JsonNode jsonNode = TransactionLogValueJsonConverter.write(transactionLogValue, version);
+                        json.set("transactionalId", new TextNode(transactionLogKey.transactionalId()));
+                        json.set("data", jsonNode);
+                    } else {
+                        json.set("data", new TextNode("unknown"));
+                    }
+                    try {
+                        output.write(json.toString().getBytes(UTF_8));
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+    }
+
+    private TransactionLogKey readToTransactionLogKey(ByteBuffer byteBuffer) {
+        short version = byteBuffer.getShort();
+        if (version >= TransactionLogKey.LOWEST_SUPPORTED_VERSION
+                && version <= TransactionLogKey.HIGHEST_SUPPORTED_VERSION) {
+            return new TransactionLogKey(new ByteBufferAccessor(byteBuffer), version);
+        } else {
+            return new TransactionLogKey();
```

Comment:
not sure whether I catch your point. I feel that we should not show the content if the version is not supported by current code. For example, users use the "old" code to parse "new" data. In that case, we should display the "unknown" instead of "default value". That rule should be applied to both key and value

Reply:
@chia7712, Thanks for your comments, If the version not in `TransactionLogValue` or `TransactionLogKey`, we should show the message
```
{
    "version": 1,
    "data": "unknown"
}
```
Only the version conform in `TransactionLogValue` and `TransactionLogKey`, and I will show 
```
{
    "version": 1,
    "transactionalId": "XXXX",
    "data": "XXX"
}
Have I misunderstood anything? Thanks for your patient.

Reply:
key and value could have different version, so it would be better to separate them into different item. For example:
```json
{
  "key": {
    "version": 1,
    "data": xxx
  },
  "value": {
    "version": 1,
    "data": xxx
  }
}
```

if we can't parse the byte array due to version unmatched, the "data" will be "unknown"

Reply:
I test the formatter, and the result is 
```json
{
  "key": {
    "version": "0",
    "data": {
      "transactionalId": "TXNID"
    }
  },
  "value": {
    "version": "1",
    "data": {
      "producerId": 100,
      "producerEpoch": 50,
      "transactionTimeoutMs": 500,
      "transactionStatus": 4,
      "transactionPartitions": [],
      "transactionLastUpdateTimestampMs": 1000,
      "transactionStartTimestampMs": 750
    }
  }
}
```
```json
{
  "key": {
    "version": "10",
    "data": "unknown"
  },
  "value": {
    "version": "10",
    "data": "unknown"
  }
}
```

Reply:
That is good and please add tests for it